skbio/tree/_nj.py

```diff
@@ -175,10 +175,11 @@ def _compute_q(dm):
     """
     q = np.zeros(dm.shape)
     n = dm.shape[0]
-    for i in range(n):
-        for j in range(i):
-            q[i, j] = q[j, i] = \
-                ((n - 2) * dm[i, j]) - dm[i].sum() - dm[j].sum()
+    dmv = dm.to_data_frame().values
```

Comment:
Just a note (I know this is merged already, maybe worth resolving in another pull request):

You can call `dm.data` to get the underlying 2D numpy array (no copying required, nor dataframe creation which can be expensive).